### ISSUE TITLE
Allow Merkle structures to be input for CLI run

### ISSUE THREAD

**Issue by vlopes11** [0xPolygonMiden/miden-vm#770]
Currently, we allow the user to construct an advice tape via the following interface

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/miden/src/cli/data.rs#L58-L70

It will read the MASM inputs file counterpart and parse sequences of `u64` into the advice tape.

However, advice providers contains Merkle structures as well, and value maps:

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/processor/src/advice/inputs.rs#L64-L78

In order to use such sets, we need to implement a way to initialize them via the inputs file. One way to achieve that is to serialize the `AdviceInputs` as json, and parse it:

https://github.com/0xPolygonMiden/miden-vm/blob/159a04f3d4648f77d664642409e521438378bd6c/processor/src/advice/inputs.rs#L17-L22

A more robust solution would be to create a full strategy to encode and decode the advice provider, probably using binary format instead of JSON.

**Comment by frisitano** [comment]
The `AdviceProvider` and associated objects are being refactored to include the new `MerkleStore` - #783.  As such I will likely have to branch off #783 or wait until it is merged into `next`.  #783 is in the merge queue waiting on #792 to land.

**Comment by bobbinth** [comment]
Closing this as loading of `MerkleTree` and `SimpleSmt` is already supported, and for loading `MerklePathSet` we have #802.
